README.md

```diff
@@ -1,4 +1,230 @@
 
+This project implements a method for faster and more memory-efficient RNN-T computation, called `pruned rnnt`.
 
-This contains an implementation of a particular recursion that turns out to be useful for fast RNN-T computation.
-We need to update this README.
\ No newline at end of file
+Note: There is also a fast RNNT loss implementation in [k2](https://github.com/k2-fsa/k2) project, which shares the same code here. We make `fast_rnnt` a stand-alone project in case someone wants only this rnnt loss.
+
+## How does the pruned-rnnt work ?
+
+We first obtain pruning bounds for the RNN-T recursion using a simple joiner network that is just an addition of the encoder and decoder, then we use those pruning bounds to evaluate the full, non-linear joiner network.
+
+The picture below display the gradients (obtained by `rnnt_loss_simple` with `return_grad=true`) of lattice nodes, at each time frame, only a small set of nodes have a non-zero gradient, which justifies the pruned RNN-T loss, i.e., putting a limit on the number of symbols per frame.
+
+<img src="https://user-images.githubusercontent.com/5284924/158116784-4dcf1107-2b84-4c0c-90c3-cb4a02f027c9.png" width="900" height="250" />
+
+> This picture is taken from [here](https://github.com/k2-fsa/icefall/pull/251)
+
+## Installation
+
+You can install it via `pip`:
+
+```
+pip install fast_rnnt
+```
+
+You can also install from source:
+
+```
+git clone https://github.com/danpovey/fast_rnnt.git
+cd fast_rnnt
+python setup.py install
+```
+
+To check that `fast_rnnt` was installed successfully, please run
+
+```
+python3 -c "import fast_rnnt; print(fast_rnnt.__version__)"
+```
+
+which should print the version of the installed `fast_rnnt`, e.g., `1.0`.
+
+
+### How to display installation log ?
+
+Use
+
+```
+pip install --verbose fast_rnnt
+```
+
+### How to reduce installation time ?
+
+Use
+
+```
+export FT_MAKE_ARGS="-j"
+pip install --verbose fast_rnnt
+```
+
+It will pass `-j` to `make`.
+
+### Which version of PyTorch is supported ?
+
+It has been tested on PyTorch >= 1.5.0.
+
+Note: The cuda version of the Pytorch should be the same as the cuda version in your environment,
+or it will cause a compilation error.
+
+
+### How to install a CPU version of `fast_rnnt` ?
+
+Use
+
+```
+export FT_CMAKE_ARGS="-DCMAKE_BUILD_TYPE=Release -DFT_WITH_CUDA=OFF"
+export FT_MAKE_ARGS="-j"
+pip install --verbose fast_rnnt
+```
+
+It will pass `-DCMAKE_BUILD_TYPE=Release -DFT_WITH_CUDA=OFF` to `cmake`.
+
+### Where to get help if I have problems with the installation ?
+
+Please file an issue at <https://github.com/danpovey/fast_rnnt/issues>
+and describe your problem there.
+
+
+## Usage
+
+### For rnnt_loss_simple
+
+This is a simple case of the RNN-T loss, where the joiner network is just
+addition.
+
+```python
+am = torch.randn((B, T, C), dtype=torch.float32)
+lm = torch.randn((B, S + 1, C), dtype=torch.float32)
+symbols = torch.randint(0, C, (B, S))
+termination_symbol = 0
```

Comment:
In RNN-T, people are usually more comfortable with the blank symbol, not the termination symbol, I think.